### DISCUSSION TITLE
Heatit Z-Water2 and Z-TRM3 Association Limit Bug in Home Assistant?

### DISCUSSION THREAD

**Discussion by krismyklebust** [zwave-js/zwave-js-ui#4072] [Q&A] [answered]
Hello everyone,

I'm experiencing an issue with Z-Wave associations in Home Assistant for my Heatit Z-Water2 device (firmware version 1.0.6) and my Heatit Z-TRM3 thermostats (firmware version 4.0.34). My Z-Wave controller is an Aeotec Z-Stick 7.
While I can successfully add three associations (in my case endpoints 5, 6, and 7 on the Z-Water2), adding a fourth one (endpoint 8) causes it not to update in Home Assistant, even though the association works physically in real life.

According to both the device documentation and the Home Assistant Z-Wave JS UI, the limit for max associations is supposed to be 5, but it seems like I’m hitting a limit at 3 associations. I've tried a different combination of endpoints to ensure there is no issue with the physical endpoint 8. 
This means the relay controller on the Z-Water2 doesn't show the toggle status in Home Assistant, making it necessary for me to physically check the actuator to see if things are actually working or not.

Has anyone else encountered this issue or have any suggestions on how to resolve this?

Thanks in advance for your help!

**Comment by robertsLando** [comment]
@krismyklebust did you tried to contact the manufacturer of the device?

cc @AlCalzone

**Comment by krismyklebust** [comment]
@robertsLando I have not, but I will get right on that. I'll report back once they reply.
Thanks so far! :)

**Comment by krismyklebust** [comment, marked as answer]
I wanted to provide an update on my previous post about the Z-Wave associations issue with my Heatit Z-Water2 and Z-TRM3 thermostats. 
After experiencing this problem for weeks, the issue has now resolved itself, and everything is working perfectly. I haven't made any changes to my setup, but it seems the problem has gone away on its own.
Thanks for the support!
